app/models/job.rb

```diff
@@ -2,10 +2,14 @@ class Job < ActiveRecord::Base
   include JobManagement
 
   belongs_to :job_template
-  has_many :steps, -> { order(:order) }, class_name: 'JobStep'
+  has_many :steps, -> { order(:order) },
+           class_name: 'JobStep',
+           dependent: :destroy
 
   serialize :environment, Array
 
+  after_destroy :destroy_job
```

Comment:
so we changed the order, right? we call destroy, which deletes from the database, then we attempt to destroy the job, right?